apps/hash-graph/lib/authorization/src/api.rs

```diff
@@ -1,36 +1,95 @@
-use std::future::Future;
+use std::{collections::HashMap, future::Future};
 
 use error_stack::Result;
-use futures::Stream;
-use graph_types::{account::AccountId, knowledge::entity::EntityId};
+use graph_types::{
+    account::{AccountGroupId, AccountId},
+    knowledge::entity::EntityId,
+    provenance::OwnedById,
+};
 
 use crate::{
-    backend::CheckError,
+    backend::{CheckError, CheckResponse},
     zanzibar::{Consistency, Zookie},
 };
 
+// TODO: Replace with something permission specific which can directly be reused once permissions
+//       are implemented.
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub enum VisibilityScope {
+    Public,
+    Account(AccountId),
+    AccountGroup(AccountGroupId),
+}
+
 pub trait AuthorizationApi {
-    async fn view_entity(
+    fn add_account_group_member(
```

Comment:
I assume these functions will take more parameters when fully implemented... e.g. the id of the member to add/remove

Reply:
Yup, they already do in the next PR 🙂